src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -99,6 +105,19 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     }
 
     /**
+     * Returns a singleton {@code HashMap}, i.e. a {@code HashMap} of one element.
+     *
+     * @param key A singleton map key.
+     * @param value A singleton map value.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given entry
+     */
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(K key, V value) {
+        return of((Comparator<? super K> & Serializable) K::compareTo, Tuple.of(key, value));
+    }
```

Comment:
Hi, please move this method behind the next method (`of(Comparator, Tuple2)`) and add an additional method

```java
public static <K, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K key, V value) {
    ...
}
```